### ISSUE TITLE
merlin broken with (include_subdirs unqualified)

### ISSUE THREAD

**Issue by nojb** [ocaml/dune#4717]
`merlin` 4.2 + OCaml 4.12 + dune 2.8 seems broken when using `(include_subdirs unqualified)`. Concretely, given a dune file in `applications/apropos/addins/dune` containing `(include_subdirs unqualified)`, when editing the file `applications/apropos/addins/models/local_vol_hull_white_model.mf` I get:
```
No config found for file "local_vol_hull_white_model.mf" in "applications\\apropos\\addins\\models". Try calling `dune build`.
```
(This is under Windows).

cc @voodoos

**Comment by nojb** [comment]
I confirm that it works OK under Unix, so this is Windows-specific.

**Comment by nojb** [comment]
I investigated the source of the problem, it is this line:
https://github.com/ocaml/dune/blob/26f5e57d1aef71f476a6dc40990b9d6a517f7849/src/dune_rules/merlin_server.ml#L115
Indeed, on Windows `path` typically contains backslashes, but `Path.Local.parent` only treats the case of forward slashes:
https://github.com/ocaml/dune/blob/26f5e57d1aef71f476a6dc40990b9d6a517f7849/otherlibs/stdune-unstable/path.ml#L242-L249

This seems like something that should not be very hard to fix. It would be great if we could get a fix in 2.9, as the current state is blocking on Windows :)

cc @voodoos
